Automattic/newspack-blocks
# HPB: fix author query if CAP is unavailable

### All Submissions:






### Changes proposed in this Pull Request:



Fixes an issue created in #1083, adds basic tests, and refactors some of the code. 

### How to test the changes in this Pull Request:

1. On `master`, disable Co-Authors Plus plugin and insert the Homepage Posts block on a page
2. Observe that querying by authors does not work (posts are not fetched)
3. Switch to this branch, observe posts are fetched

### Other information:
